YouTube.js

```diff
@@ -89,14 +89,18 @@ function doWeb(doc, url) {
 
 function scrape(doc, url) {
 	var item = new Zotero.Item("videoRecording");
+
+	/* YouTube won't update the meta tags for the user,
+	 * if they open e.g. a suggested video in the same tab.
+	 * Thus we scrape them from screen instead.
+	 */
+
 	item.title = text(doc, '#info-contents h1.title');
-	item.url = url;
+	// try to scrape only the canonical url, excluding additional query parameters
+	item.url = url.replace(/^(.+\/watch\?\bv=[0-9a-zA-Z_-]+).*/, "$1");
```

Comment:
Why the `\b`? And won't this not work on valid URLs like https://www.youtube.com/watch?app=desktop&v=HfIymdQ2LWc?